features/bsm/daemon/src/main/java/org/opennms/netmgt/bsm/daemon/Bsmd.java

```diff
@@ -0,0 +1,310 @@
+import org.opennms.netmgt.config.api.EventConfDao;
+import org.opennms.netmgt.daemon.SpringServiceDaemon;
+import org.opennms.netmgt.dao.api.AlarmDao;
+import org.opennms.netmgt.events.api.EventConstants;
+import org.opennms.netmgt.events.api.EventIpcManager;
+import org.opennms.netmgt.events.api.annotations.EventHandler;
+import org.opennms.netmgt.events.api.annotations.EventListener;
+import org.opennms.netmgt.model.OnmsAlarm;
+import org.opennms.netmgt.model.OnmsSeverity;
+import org.opennms.netmgt.model.events.EventBuilder;
+import org.opennms.netmgt.xml.event.Event;
+import org.opennms.netmgt.xml.event.Parm;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.TransactionStatus;
+import org.springframework.transaction.support.TransactionCallbackWithoutResult;
+import org.springframework.transaction.support.TransactionTemplate;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * This daemon is responsible for driving the Business Service state machine by:
+ *  - Sending alarms to the state machine when they are created and/or updated
+ *  - Sending events on the event bus when the state of a Business Service changes
+ *  - Reloading the Business Service configuration in the state machine when requested
+ *
+ * Some caveats to keep in account:
+ *  - We may not always receive the alarm life-cycle events, we will need to poll the database periodically.
+ *  - Events we do receive may happen out-of-order, i.e. alarm created, alarm deleted, alarm escalated
+ *
+ * @author jwhite
+ */
+@EventListener(name=Bsmd.NAME, logPrefix="bsmd")
+public class Bsmd implements SpringServiceDaemon, BusinessServiceStateChangeHandler  {
+    private static final Logger LOG = LoggerFactory.getLogger(Bsmd.class);
+
+    public static final String NAME = "Bsmd";
+
+    @Autowired
+    private BusinessServiceDao m_businessServiceDao;
+
+    @Autowired
+    private AlarmDao m_alarmDao;
+
+    @Autowired
+    private EventIpcManager m_eventIpcManager;
+
+    @Autowired
+    private EventConfDao m_eventConfDao;
+
+    @Autowired
+    private TransactionTemplate m_template;
+
+    @Autowired
+    private BusinessServiceStateMachine m_stateMachine;
+
+    private boolean m_verifyReductionKeys = true;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        Preconditions.checkNotNull(m_stateMachine, "stateMachine cannot be null");
+
+        LOG.info("Initializing bsmd...");
+        m_stateMachine.addHandler(this, null);
+    }
+
+    @Override
+    public void start() throws Exception {
+        Preconditions.checkNotNull(m_businessServiceDao, "businessServiceDao cannot be null");
+        Preconditions.checkNotNull(m_alarmDao, "alarmDao cannot be null");
+        Preconditions.checkNotNull(m_eventIpcManager, "eventIpcManager cannot be null");
+        Preconditions.checkNotNull(m_eventConfDao, "eventConfDao cannot be null");
+
```

Comment:
Just a note. If we use Objects.requireNonNull(...) than we can remove the dependency to guava.